metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/DefaultLogPublisher.java

```diff
@@ -266,34 +294,65 @@ protected void printProducerLog(DefaultPublishModel model, StringBuilder sb) {
       }
     }
     //print details
-    sb.append("  details:\n")
-        .append(detailsBuilder);
+    sb.append("  details:\n").append(detailsBuilder);
   }
 
-
   private StringBuilder printSamplePerf(OperationPerfGroup perfGroup) {
     StringBuilder sb = new StringBuilder();
     for (int i = 0; i < perfGroup.getOperationPerfs().size(); i++) {
       OperationPerf operationPerf = perfGroup.getOperationPerfs().get(i);
       PerfInfo stageTotal = operationPerf.findStage(MeterInvocationConst.STAGE_TOTAL);
+      //may be null
+      OperationLatencyDistribution operationLatencyDistribution = operationPerf.getLatencyDistribution();
       if (i == 0) {
         // first line
         String status = perfGroup.getTransport() + "." + perfGroup.getStatus();
-        sb.append(String.format(FIRST_LINE_SIMPLE_FORMAT, status, stageTotal.getTps(),
-            getDetailsFromPerf(stageTotal), operationPerf.getOperation()));
+        sb.append(String.format(FIRST_LINE_SIMPLE_FORMAT, status,
+            stageTotal.getTps(),
+            getDetailsFromPerf(stageTotal),
+            getSortedStrFromPerf(operationLatencyDistribution),
+            operationPerf.getOperation()));
       } else {
-        sb.append(String
-            .format(SIMPLE_FORMAT, stageTotal.getTps(), getDetailsFromPerf(stageTotal), operationPerf.getOperation()));
+        sb.append(String.format(SIMPLE_FORMAT, stageTotal.getTps(),
+            getDetailsFromPerf(stageTotal),
+            getSortedStrFromPerf(operationLatencyDistribution),
+            operationPerf.getOperation()));
       }
     }
-    //print summary
     OperationPerf summaryOperation = perfGroup.getSummary();
     PerfInfo stageSummaryTotal = summaryOperation.findStage(MeterInvocationConst.STAGE_TOTAL);
-    sb.append(
-        String.format(SIMPLE_FORMAT, stageSummaryTotal.getTps(), getDetailsFromPerf(stageSummaryTotal), "(summary)"));
+    OperationLatencyDistribution latencyDistributionSummary = summaryOperation.getLatencyDistribution();
+    //print summary
+    sb.append(String.format(SIMPLE_FORMAT, stageSummaryTotal.getTps(),
+        getDetailsFromPerf(stageSummaryTotal),
+        getSortedStrFromPerf(latencyDistributionSummary),
+        "(summary)"));
     return sb;
   }
 
+  private String getSortedStrFromPerf(OperationLatencyDistribution operationLatencyDistribution) {
+    if (operationLatencyDistribution == null || sortedLatencyIntervals == null) {
+      return "";
+    }
+    Object[] array = sortedLatencyIntervals.stream()
+        .map(value -> operationLatencyDistribution.getInterval().getOrDefault(value, 0))
+        .toArray();
+    return String.format(intervalFormat, array);
+  }
+
+  public String getLogStrFormat(List<String> latencyIntervals) {
```

Comment:
1.private
2.change name relate to latency distribution

Reply:
done. the new method is as bellow 
```java
  /**
   * <p>
   *     get respond str format from list of str. for example
   *    {@code
   *         List<String> strList = Arrays.asList("[0,1) ", "[1,4) ", "[4, ) ");
   *         String strFormat = getLatencyScopeFormat(strList);
   *         System.out.println(String.format(strFormat,strList.toArray()));
   *         System.out.println(String.format(strFormat, 1, 23, 433));
   *         System.out.println(String.format(strFormat, 121, 2, 4));
   *    }
   *    In the example above, result is showed as bellow:
   *        [0,1)  [1,4)  [4, )
   *        1      23     433
   *        121    2      4
   * </p>
   * @param latencyIntervals list of str
   * @return str format
   */
  private String getLatencyScopeFormat(List<String> latencyIntervals) {
```